app/src/main/java/com/restart/spacestationtracker/Preferences.java

```diff
@@ -158,7 +158,7 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String permissi
          * @return True if permission is granted. False if otherwise.
          */
         public boolean isLocationPermissionGranted() {
-            return Build.VERSION.SDK_INT >= 23 && mContext.checkSelfPermission(android.Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED;
+            return Build.VERSION.SDK_INT <= Build.VERSION_CODES.M || mContext.checkSelfPermission(android.Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED;
```

Comment:
Good catch, but I think it needs to be just `<` because Marshmallow (23) requires the run time permission.

Reply:
Also just for why it was happening, the logic was indeed incorrect. We were requesting permissions for API below 23, then calling `getLocationPermission()` which is a method targeting >= 23, so we would get this stack.

```
    java.lang.NoSuchMethodError: No virtual method requestPermissions([Ljava/lang/String;I)V in class Lcom/restart/spacestationtracker/Preferences$SettingsFragment; or its super classes (declaration of 'com.restart.spacestationtracker.Preferences$SettingsFragment' appears in /data/app/com.restart.spacestationtracker-2/split_lib_slice_2_apk.apk)
        at com.restart.spacestationtracker.Preferences$SettingsFragment.getLocationPermission(Preferences.java:131)
        at com.restart.spacestationtracker.Preferences$SettingsFragment.onISSNotification(Preferences.java:259)
        at com.restart.spacestationtracker.Preferences$SettingsFragment.onPreferenceClick(Preferences.java:203)
```

Reply:
Yeah exactly kiarasht. Thanks.

Reply:
If you fix the [first comment I gave](https://github.com/Kiarasht/Space-Station-Tracker/pull/4#discussion_r187788676) I think we can merge this then.